src/pages/SignUpPage.jsx

```diff
@@ -2,17 +2,17 @@ import InnerContainer from "@/components/shared/InnerContainer";
 import SignUp from "@/components/shared/SignUp";
 import { getRoles } from "@/lib/apiService";
 import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchRolesIfNeeded } from "@/clientThunks";
 
 function SignUpPage() {
-  const [roles, setRoles] = useState([]);
+  const dispatch = useDispatch();
+  const roles = useSelector((state) => state.clientReducer.roles);
+
   useEffect(() => {
-    getRoles()
-      .then((response) => {
-        setRoles(response.data);
-        console.log("data", response.data);
-      })
-      .catch((error) => console.error(error));
-  }, []);
+    dispatch(fetchRolesIfNeeded());
+  }, [dispatch]);
```

Comment:
dispatch dependency olarak useEffect içinde gereksiz. Boş bir array [] yeterli olur, dispatch referansı değişmediği için tekrar çağrılmaya gerek kalmaz.